New feature - 3. 3.1 As admin (don't worry he is a simple user) I wan…

scripa01 opened the pull request:
…t to have statistic data about difference  between creation date and receive date. I want to have an API to get next data:

                 ORDERID, USERID, USERNAME, ORDERED DATE, RECEIVED DATE, DIFFERENCE (in days), ORDER STATUS, list of product ids
                 + Bug fixes.

scripa01 committed:
New feature - 3. 3.1 As admin (don't worry he is a simple user) I want to have statistic data about difference  between creation date and receive date. I want to have an API to get next data:
                 ORDERID, USERID, USERNAME, ORDERED DATE, RECEIVED DATE, DIFFERENCE (in days), ORDER STATUS, list of product ids
                 + Bug fixes.

ngherta commented on src/main/java/com/example/springexploring/controller/OrderStatisticController.java:
```diff
@@ -0,0 +1,29 @@
+        return ResponseEntity.ok(orderStatisticService.findAll());
+    }
+
+    @GetMapping("/{id}")
```
Bad naming. User id, item id?

ngherta commented on src/main/java/com/example/springexploring/dto/Mapper/OrderStatisticMapper.java:
```diff
@@ -0,0 +1,31 @@
+import java.util.stream.Collectors;
+
+@Component
+@RequiredArgsConstructor
```
Redundant. We don't have any dependencies.

ngherta commented on src/main/java/com/example/springexploring/service/OrderServiceImpl.java:
```diff
@@ -83,4 +85,9 @@ private void assertThatUserAreOwner(Long userId, Order order) {
             throw new CustomRuntimeException("User with Id - " + userId + " dont have order with Id - " + order.getId());
     }
 
+    private void assertThatUserAlreadyReceivedOrder(Order order) {
```
assertThatOrderIsNotReceived
You don't check that user receives order. It's scenario when we throw exception.

ngherta commented on src/main/java/com/example/springexploring/service/OrderStatisticService.java:
```diff
@@ -0,0 +1,12 @@
+public interface OrderStatisticService {
+    List<OrderStatisticDTO> findAll();
+
+    OrderStatisticDTO findById(Long id);
```
What kind of ID?
    OrderStatisticDTO findById(Long orderId);

scripa01 committed:
New feature - 3. 3.1 As admin (don't worry he is a simple user) I want to have statistic data about difference  between creation date and receive date. I want to have an API to get next data:
                 ORDERID, USERID, USERNAME, ORDERED DATE, RECEIVED DATE, DIFFERENCE (in days), ORDER STATUS, list of product ids
                 + Bug fixes.

scripa01 committed:
New feature - 3. 3.1 As admin (don't worry he is a simple user) I want to have statistic data about difference  between creation date and receive date. I want to have an API to get next data:
                 ORDERID, USERID, USERNAME, ORDERED DATE, RECEIVED DATE, DIFFERENCE (in days), ORDER STATUS, list of product ids
                 + Bug fixes.
